extensions/amp-experiment/1.0/mutation-parser.js

```diff
@@ -65,23 +36,35 @@ const SUPPORTED_ATTRIBUTES = {
  * @return {!Function}
  */
 export function parseMutation(mutation, document) {
-  const mutationRecord = assertMutationRecord(mutation);
+  const mutationRecord = assertMutationRecordFormat(mutation);
 
   const stringifiedMutation = JSON.stringify(mutation);
 
   setSelectorToElement('target', mutationRecord, document);
 
   if (mutationRecord['type'] === 'attributes') {
-    assertAttributeMutation(mutationRecord, stringifiedMutation);
+    assertAttributeMutationFormat(mutationRecord, stringifiedMutation);
 
-    return () => {
-      mutationRecord['targetElement'].setAttribute(
-        mutationRecord['attributeName'],
-        mutationRecord['value']
-      );
-    };
+    // Get the AttribeMutationEntry
+    // From the Attribute allow list
+    const allowedAttributeMutationEntry = getAllowedAttributeMutationEntry(
+      mutationRecord,
+      stringifiedMutation
+    );
+
+    // Assert the attribute mutation passes it's check
+    // that is allowed and validated.
+    userAssert(
+      allowedAttributeMutationEntry.validate(mutationRecord['value']),
+      'Mutation %s has an an unsupported value.',
+      stringifiedMutation
+    );
+
+    // Return the corresponding mutation
+    // For the allowed attribute mutation
+    return allowedAttributeMutationEntry.mutate.bind(this, mutationRecord);
   } else if (mutationRecord['type'] === 'characterData') {
-    assertCharacterDataMutation(mutationRecord, stringifiedMutation);
+    assertCharacterDataMutationFormat(mutationRecord, stringifiedMutation);
 
     return () => {
```

Comment:
We've talked about this offline. If we are going to add `innerHTML` and `reorder` mutation, I think we will need to unify the format. Maybe create an interface or class for mutationEntry that `allowedAttributeMutationEntry` implement. 
We don't need to worry about that yet, but could you please leave a comment that this is a mutationEntry where validate() is noop.